stacker/plan.py

```diff
@@ -224,9 +230,15 @@ def walk_func(fqn):
         if self._reverse:
             dag = dag.transpose()
 
-        return dag.walk(walk_func)
+        walk = dag.walk
+        if parallel:
+            walk = dag.walk_parallel
+
+        return walk(walk_func, cancel=cancel)
 
     def _check_point(self):
+        lock = self._check_point_lock
```

Comment:
I think this probably needs to be extended to lock calls to `fn` in `execute`. I ran into a case where I was using an aws profile with MFA, and got hit with 20+ prompts at once. Individual `fn` calls should be pretty quick, so holding the lock shouldn't be too bad.